CONTRIBUTING.adoc

```diff
@@ -2,45 +2,135 @@
-* `cabal-install`
-* Agda with the right libraries set up to build our Agda code
-* `stylish-haskell`
-* `haskell-language-server`
-* `editorconfig-checker`
-* ... and more
+`error: experimental Nix feature 'nix-command' is disabled; use '--extra-experimental-features nix-command' to override`
 
-We rely heavily on this approach to ensure that everyone has consistent versions of the tools that we use.
-Please do make use of this, since problems due to mismatched versions of tools are particularly annoying to fix!
+Then try this:
 
-NOTE: You may want to consider using https://github.com/target/lorri[lorri] as a convenient alternative to running `nix develop` directly.
+`nix develop --extra-experimental-features 'nix-command flakes'`
 
-=== How to use helper scripts to fix some common issues
+You may now choose to edit your `/etc/nix/nix.conf` (global) or `~/.config/nix/nix.conf` (you must be a https://nixos.org/nix/manual/#ssec-multi-user[trusted user] to do this) by appending the following line: 
 
-The shell comes with some tools for fixing various simple problems that the CI will complain about. Specifically:
+`extra-experimental-features = nix-command flakes`
 
-- `fix-stylish-haskell` will re-format all the Haskell sources correctly.
-- `fix-cabal-fmt` will re-format all the Cabal sources correctly.
-- `fix-png-optimization` will optimize all PNGs in the repository.
+If you do so you will be able to enter the development shell by running `nix develop`.
 
-If you're not using `nix develop` all the time and you want to run one of these, you can use `nix develop --command`.
+If you are a VSCode user, you may want to start your development session by running:
+
+`nix develop --command code`
+
+If this is your first time running `nix develop`, you may be asked the following:
+```
+do you want to allow configuration setting 'allow-import-from-derivation' to be set to 'true' (y/N)? y
+do you want to permanently mark this value as trusted (y/N)? y
+```
+Please type `y` to this and all subsequent promopts and make life easy for yourself.
+
+IMPORTANT: *It is particulary important to accept the `extra-subsitures` settings, which will grant access to our Nix binary cache.*
+
+At this point and if this is your first time running `nix develop`, Nix will download, build and install a copious amout of dependencies.
+
+It is not uncommon for this process to take a couple of hours and to write tens of GBs to the `/nix/store`.
+
+*Rest assured that this only happens the very first time you run `nix develop`.*
+
+However you should pay attention to the output of `nix develop`: if you notice that you are *building* GHC, it is likely that your 
+caches have not been configured correctly.
+
+Accepting the configuration settings as outlined above should be suffienct to avoid this, however if your caches are still broken, you may try to append the following lines to `/etc/nix/nix.conf` (global) and/or `~/.config/nix/nix.conf` (user):
```

Comment:
```suggestion
Accepting the configuration settings as outlined above should be sufficient to avoid this, however if your caches are still broken, you may try to append the following lines to `/etc/nix/nix.conf` (global) and/or `~/.config/nix/nix.conf` (user):
```